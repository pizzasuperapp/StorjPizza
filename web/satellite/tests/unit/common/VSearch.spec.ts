// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import SearchComponent from '@/components/common/VSearch.vue';

import { mount, shallowMount } from '@vue/test-utils';

describe('SearchComponent.vue', () => {
    it('renders correctly', () => {
        const wrapper = shallowMount<SearchComponent>(SearchComponent);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with default props', () => {
        const wrapper = mount<SearchComponent>(SearchComponent);

        expect(wrapper.vm.$props.placeholder).toMatch('');
    });

    it('functions onMouseEnter/onMouseLeave work correctly', () => {
        const wrapper = mount<SearchComponent>(SearchComponent);

        wrapper.vm.onMouseEnter();

        expect(wrapper.vm.style.width).toMatch('540px');

        wrapper.vm.onMouseLeave();

        expect(wrapper.vm.searchString).toMatch('');
        expect(wrapper.vm.style.width).toMatch('56px');
    });

    it('function clearSearch works correctly', () => {
        const processSearchQuerySpy = jest.fn();

        const wrapper = mount<SearchComponent>(SearchComponent);

        wrapper.vm.processSearchQuery = processSearchQuerySpy;
        wrapper.vm.clearSearch();

        expect(processSearchQuerySpy).toHaveBeenCalledTimes(1);
        expect(wrapper.vm.$data.inputWidth).toMatch('56px');
    });
});
