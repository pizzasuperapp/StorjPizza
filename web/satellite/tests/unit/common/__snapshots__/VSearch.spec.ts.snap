// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent.vue renders correctly 1`] = `<input placeholder="Search " type="text" autocomplete="off" class="common-search-input" style="width: 56px;">`;
